catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityMarkerPlugin.java

```diff
@@ -0,0 +1,142 @@
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PreIngestPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.util.Describable;
+import ddf.catalog.validation.MetacardValidator;
+import ddf.catalog.validation.ValidationException;
+
+public class MetacardValidityMarkerPlugin implements PreIngestPlugin {
+
+    private List<String> enforcedMetacardValidators;
+
+    private List<MetacardValidator> metacardValidators;
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(MetacardValidityMarkerPlugin.class);
+
+    public static final String NONE = "none";
+
+    public static final String VALIDATION_ERRORS = "validation-errors";
+
+    public static final String VALIDATION_WARNINGS = "validation-warnings";
+
+    @Override
+    public CreateRequest process(CreateRequest input)
+            throws PluginExecutionException, StopProcessingException {
+
+        // Initialize empty list of metacards to allow through
+        List<Metacard> returnMetacards = new ArrayList<Metacard>();
+        // Loop all metacards in request
+        for (Metacard metacard : input.getMetacards()) {
+            MetacardImpl newMetacard = new MetacardImpl(metacard);
+            List<Serializable> validationWarnings = new LinkedList<>();
+            List<Serializable> validationErrors = new LinkedList<>();
+            // Default to allowing metacard through
+            Boolean toReturn = true;
+            // Run metacard through each validator
+            for (MetacardValidator metacardValidator : metacardValidators) {
+                try {
+                    // Attempt validation
+                    metacardValidator.validate(metacard);
+                } catch (ValidationException e) {
+                    if (metacardValidator instanceof Describable) {
+                        // If validator is not explicitly turned on by admin, set invalid and allow through
+                        if (checkEnforcedMetacardValidators(metacardValidator)) {
+                            // Check for warnings and errors
+                            if (!e.getErrors().isEmpty()) {
+                                validationErrors.add(((Describable) metacardValidator).getId());
+                            }
+                            if (!e.getWarnings().isEmpty()) {
+                                validationWarnings.add(((Describable) metacardValidator).getId());
+                            }
+                        } else {
+                            // If validator is explicitly turned on, break out and do not include in the
+                            // list of metacards that will be allowed through.
+                            toReturn = false;
+                            break;
+                        }
+                    } else {
+                        // TODO: Do we want validator plugin to fail when a failed validator doesn't implement describable?
```

Comment:
You could extract that logic into a method to get an ID for a validator.  Then it can be reused in the `checkEnforacedMetacardValidators` method as well.